packages/grafana-data/src/field/scale.ts

```diff
@@ -1,127 +1,63 @@
-import { Field, Threshold, GrafanaTheme, GrafanaThemeType, ThresholdsMode, FieldColorMode } from '../types';
-import { reduceField, ReducerID } from '../transformations';
-import { getColorFromHexRgbOrName } from '../utils/namedColorsPalette';
-import * as d3 from 'd3-scale-chromatic';
-import isNumber from 'lodash/isNumber';
+import { isNumber } from 'lodash';
+import { reduceField, ReducerID } from '../transformations/fieldReducer';
+import { Field, GrafanaTheme, Threshold } from '../types';
+import { getFieldColorModeFor } from './fieldColor';
+import { getActiveThresholdForValue } from './thresholds';
 
 export interface ScaledValue {
-  percent?: number; // 0-1
-  threshold?: Threshold; // the selected step
-  color?: string; // Selected color (may be range based on threshold)
+  percent: number; // 0-1
+  threshold: Threshold;
+  color: string;
 }
 
 export type ScaleCalculator = (value: number) => ScaledValue;
 
-/**
- * @param t Number in the range [0, 1].
- */
-type colorInterpolator = (t: number) => string;
+export function getScaleCalculator(field: Field, seriesIndex: number, theme: GrafanaTheme): ScaleCalculator {
```

Comment:
@torkelo I see the `seriesIndex` is now being passed around in few places. What do you think about introducing `seriesIndex` in the `FieldState` instead? Think this would be easier to support other cases where the series index is interesting.

Reply:
interesting idea! think that should work.

Reply:
updates to use state.seriesIndex , only a bit wonky in the graph3 panel as it is using a transform, so have to update seriesIndex before calling the scale getScaleCalculator function. But otherwise it worked out gret